src/box/lua/tuple.lua

```diff
@@ -65,6 +67,30 @@ local builtin = ffi.C
 local tuple_t = ffi.typeof('box_tuple_t')
 local const_tuple_ref_t = ffi.typeof('box_tuple_t&')
 
+local new_tuple = function(...)
+    if compat.box_tuple_new_arg_list.current == 'old' then
+        return internal.tuple.new(...)
+    end
+    local argc = select('#', ...)
+    if argc <= 1 then
+        return internal.tuple.new(...)
+    elseif argc > 2 then
+        error("Usage: box.tuple.new({tuple-fields}, {options})")
+    else
+        local tuple, options = ...
+        if type(options) ~= 'table' then
+            error("box.tuple.new(): second argument (options) must be a table")
+        end
+        if options.format == nil then
+            return internal.tuple.new(tuple)
+        end
+        if type(options.format) ~= 'cdata' then
+            options.format = box.tuple.format.new(options.format)
```

Comment:
It can a bit confuse. This function could change user input.

This lead that users create issues like that https://github.com/tarantool/tarantool/issues/3853

Reply:
Fixed, added a test case. Thanks!